src/Workspaces/CSharp/Portable/Extensions/SemanticModelExtensions.cs

```diff
@@ -445,5 +477,103 @@ private static TypeSyntax GetOutermostType(TypeSyntax type)
         {
             return type.GetAncestorsOrThis<TypeSyntax>().Last();
         }
+
+
+        // API that  gets semantic info for a specific arguments from @dustinca
+        //https://github.com/DustinCampbell/CSharpEssentials/blob/master/Source/CSharpEssentials/Extensions.cs
+        // supposedly there is a bug on github somewhere for him to add this api to the compiler
+        internal struct ArgumentInfo
+        {
+            public readonly ISymbol MethodOrProperty;
+            public readonly IParameterSymbol Parameter;
+
+            public ArgumentInfo(ISymbol methodOrProperty, IParameterSymbol parameter)
+            {
+                this.MethodOrProperty = methodOrProperty;
+                this.Parameter = parameter;
+            }
+        }
+
+        public static ArgumentInfo GetArgumentInfo(this SemanticModel semanticModel, ArgumentSyntax argument)
+        {
+            if (semanticModel == null)
+            {
+                throw new System.ArgumentNullException(nameof(semanticModel));
+            }
+
+            if (argument == null)
+            {
+                throw new System.ArgumentNullException(nameof(argument));
+            }
+
+            var argumentList = argument.Parent as ArgumentListSyntax;
+            if (argumentList == null)
+            {
+                return default(ArgumentInfo);
+            }
+
+            var expression = argumentList.Parent as ExpressionSyntax;
+            if (expression == null)
+            {
+                return default(ArgumentInfo);
+            }
+
+            var methodOrProperty = semanticModel.GetSymbolInfo(expression).Symbol;
+            if (methodOrProperty == null)
+            {
+                return default(ArgumentInfo);
+            }
+
+            var parameters = methodOrProperty.GetParameters();
+            if (parameters.Length == 0)
+            {
+                return default(ArgumentInfo);
+            }
+
+            if (argument.NameColon != null)
+            {
+                if (argument.NameColon.Name == null)
+                {
+                    return default(ArgumentInfo);
+                }
+
+                // We've got a named argument...
+                var nameText = argument.NameColon.Name.Identifier.ValueText;
+                if (nameText == null)
+                {
+                    return default(ArgumentInfo);
+                }
+
+                foreach (var parameter in parameters)
+                {
+                    if (string.Equals(parameter.Name, nameText, System.StringComparison.Ordinal))
```

Comment:
We should see if we can write this in a language agnostic fashion.  Casey was asking me about IOperation, and this would be a good example of how you can use IOperation to write code across VB/C# in a uniform manner for cases like these where the languages are practically the same.